scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py

```diff
@@ -382,10 +447,28 @@ def test_random_initial_float32():
     A = A.astype(np.float32)
     X = rnd.random((n, m))
     X = X.astype(np.float32)
```

Comment:
corresponding to types suggestion for signature above
```suggestion
    X = rnd.random((n, m))
    if np.issubdtype(dtype, np.complexfloating):
        # populate imaginary values for A & X as well
        A += 1j * A
        X += 1j * X
    A = A.astype(dtype)
    X = X.astype(dtype)
```
Not sure if we want to mix matrices for the imaginary component to avoid that the real & imaginary components are linearly dependent, but I guess the above just rotates all eigenvalues by `(sqrt(2), pi/2)`.

Reply:
`A` must be Hermitian in LOBPCG, so its diagonal must be real.